systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/externalClients/OauthExternalKafkaClient.java

```diff
@@ -0,0 +1,273 @@
+        String clientName = "receiver-plain-" + clusterName;
+        vertx = Vertx.vertx();
+        CompletableFuture<Integer> resultPromise = new CompletableFuture<>();
+
+        IntPredicate msgCntPredicate = x -> x == messageCount;
+
+        KafkaClientProperties kafkaClientProperties = new KafkaClientProperties.KafkaClientPropertiesBuilder()
+            .withNamespaceName(namespaceName)
+            .withClusterName(clusterName)
+            .withGroupIdConfig(consumerGroup)
+            .withSecurityProtocol("SASL_" + CommonClientConfigs.DEFAULT_SECURITY_PROTOCOL)
+            .withBootstrapServerConfig(getExternalBootstrapConnect(namespaceName, clusterName))
+            .withKeyDeSerializerConfig(StringDeserializer.class.getName())
+            .withValueDeSerializerConfig(StringDeserializer.class.getName())
+            .withClientIdConfig(kafkaUsername + "-consumer")
+            .withAutoOffsetResetConfig("earliest")
+            .withSaslMechanism("OAUTHBEARER")
+            .withSaslLoginCallbackHandlerClass()
+            .withSharedProperties()
+            .withSaslJassConfig(this.clientId, this.clientSecretName, this.oauthTokenEndpointUri)
+            .build();
+
+        vertx.deployVerticle(new Consumer(kafkaClientProperties, resultPromise, msgCntPredicate, topicName, clientName));
+
+        try {
+            resultPromise.get(timeoutMs, TimeUnit.MILLISECONDS);
+        } catch (Exception e) {
+            resultPromise.completeExceptionally(e);
+        }
+        vertx.close();
+        return resultPromise;
+    }
+
+    public Future<Integer> receiveMessagesTls() {
+        return sendMessagesTls(Constants.GLOBAL_CLIENTS_TIMEOUT);
+    }
+
+    @Override
+    public Future<Integer> receiveMessagesTls(long timeoutMs) {
+
+        String clientName = "receiver-ssl-" + clusterName;
+        vertx = Vertx.vertx();
+        CompletableFuture<Integer> resultPromise = new CompletableFuture<>();
+
+        IntPredicate msgCntPredicate = x -> x == messageCount;
+
+        String caCertName = this.caCertName == null ?
+                KafkaResource.getKafkaExternalListenerCaCertName(namespaceName, clusterName) : this.caCertName;
+        LOGGER.info("Going to use the following CA certificate: {}", caCertName);
+
+        KafkaClientProperties kafkaClientProperties = new KafkaClientProperties.KafkaClientPropertiesBuilder()
+            .withNamespaceName(namespaceName)
+            .withClusterName(clusterName)
+            .withCaSecretName(caCertName)
+            .withBootstrapServerConfig(getExternalBootstrapConnect(namespaceName, clusterName))
+            .withKeyDeSerializerConfig(StringDeserializer.class.getName())
+            .withValueDeSerializerConfig(StringDeserializer.class.getName())
+            .withKafkaUsername(kafkaUsername)
+            .withSecurityProtocol("SASL_SSL")
+            .withGroupIdConfig(consumerGroup)
+            .withAutoOffsetResetConfig("earliest")
```

Comment:
Similarly you could use `org.apache.kafka.clients.consumer.OffsetResetStrategy`

Reply:
This will be questioned for Serialization and also this one. What gives me that I can provide `OffsetResetStrategy.class`? It is just of type compatibility and the wreck of more exuberance of semanticism?  I totally agree it is better to use classes instead of literals like this, but from the performance POV it is easy to just provide string right not just whole class?

Reply:
I think you misunderstood my comment. The argument provided by the caller in this example would be `OffsetResetStrategry.EARLIEST`. For the serializer you'd need to provide a `Class<? extends Serializer>`. Different cases of the same general point, which is that it's not very type safe to use Strings for these things, and there are existing types you can easily use to provide more type safety to your API.

Reply:
Yeah, got it. Maybe last question using of`Class<? extends Serializer>` would be using this class in the raw. Would it be still ok instead of specifying concrete type Serializer<T>. In that case `T will be StringSerializer` i assume